packages/block-library/src/section/editor.scss

```diff
@@ -0,0 +1,61 @@
+/**
+ * Section: All Alignment Settings
+ */
+.wp-block[data-type="core/section"] {
+	@include break-small() {
+		padding-left: $block-padding*3;
+		padding-right: $block-padding*3;
+	}
+
+	// Only applied when background is added to cancel out padding
+	> .editor-block-list__block-edit > div > .wp-block-section.has-background > .editor-inner-blocks {
+		margin-top: -#{$block-padding*2 + $block-spacing};
+		margin-bottom: -#{$block-padding*2 + $block-spacing};
+	}
+
+	// Full Width Blocks
+	// specificity required to only target immediate child Blocks of Section
+	> .editor-block-list__block-edit > div > .wp-block-section > .editor-inner-blocks > .editor-block-list__layout > .wp-block[data-align="full"] {
+		margin-left: auto;
+		margin-right: auto;
+		padding-left: $block-padding*2;
+		padding-right: $block-padding*2;
+
+		@include break-small() {
+			padding-left: $block-padding*4 + $block-spacing/2; // 58px
+			padding-right: $block-padding*4 + $block-spacing/2; // 58px
+		}
+	}
+}
+
+/**
+ * Section: Full Width Alignment
+ */
+.wp-block[data-type="core/section"][data-align="full"] {
+
+	// First tier of InnerBlocks must act like the container of the standard canvas
+	> .editor-block-list__block-edit > div > .wp-block-section > .editor-inner-blocks {
+		margin-left: auto;
+		margin-right: auto;
+		padding-left: $block-padding*3 + $block-spacing + $border-width;
+		padding-right: $block-padding*3 + $block-spacing + $border-width;
+	}
+
+	// Full Width Blocks
+	// specificity required to only target immediate child Blocks of Section
+	> .editor-block-list__block-edit > div > .wp-block-section > .editor-inner-blocks > .editor-block-list__layout > .wp-block[data-align="full"] {
```

Comment:
One note on the specificity here: Not only do we need this for Gutenberg's purposes here, but if themes need to override these styles (Twenty Nineteen does for instance), they'll also need to use this specificity. Not totally ideal, but I'm not sure there's a great way around it.

Reply:
It's the only way I found to ensure you're only targetted the first "level" of child Blocks. Else it gets really messy. Open to alternatives...